productivity-suite/piercing-library/src/message-bus/message-bus.ts

```diff
@@ -0,0 +1,54 @@
+export type MessageHandler = {
+  eventName: string;
+  callback: (detail: any) => void;
+  options?: { once?: boolean };
+};
+
+export type MessageBusState = Record<string, { detail: any }>;
+
+export interface MessageBus {
+  state: MessageBusState;
+  dispatch(eventName: string, detail: any): void;
+  listen(handler: MessageHandler): (() => void) | null;
+}
+
+export class GenericMessageBus implements MessageBus {
+  protected _handlers: MessageHandler[] = [];
+
+  constructor(protected _state: MessageBusState = {}) {}
+
+  get state(): MessageBusState {
+    return JSON.parse(JSON.stringify(this._state));
```

Comment:
overkill? 🤔

Reply:
@petebacondarwin with your new types you've suggested the fact that the state is serializable is guaranteed by typescript (unless someone hacks around).

So do you still agree to keep this in? 🙂

Reply:
Yeah let's keep this in for now.